contracts/schemes/JoinAndQuit.sol

```diff
@@ -0,0 +1,243 @@
+pragma solidity ^0.5.16;
+
+import "../votingMachines/VotingMachineCallbacks.sol";
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+import "../libs/StringUtil.sol";
+
+
+/**
+ * @title A scheme for join in a dao.
+ * - A member can be proposed to join in by sending a min amount of fee.
+ * - A member can ask to quite (RageQuit) a dao on any time.
+ * - A member can donate to a dao.
+ */
+contract JoinAndQuit is
+        VotingMachineCallbacks,
+        ProposalExecuteInterface,
+        Initializable {
+    using SafeMath for uint;
+    using SafeERC20 for address;
+    using StringUtil for string;
+
+    event JoinInProposal(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        string _descriptionHash,
+        address _proposedMember,
+        uint256 _fundAmount
+    );
+
+    event FundedDeadLineReached(
+        address indexed _avatar
+    );
+
+    event Donation(
+        address indexed _avatar,
+        uint256 indexed _donation
+    );
+
+    event RageQuit(
+        address indexed _avatar,
+        uint256 indexed _refund
+    );
+
+    event RedeemReputation(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount);
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _decision);
+
+    struct Proposal {
+        bool accepted;
+        address proposedMember;
+        address funder;
+        uint256 funding;
+    }
+
+    mapping(bytes32=>Proposal) public proposals;
+    mapping(address=>uint256) public fundings;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    Avatar public avatar;
+    IERC20 public fundingToken;
+    uint256 public minFeeToJoin;
+    uint256 public memberReputation;
+    uint256 public fundingGoal;
```

Comment:
Other schemes may want to access this data. With the current logic, they can only see if the goal has been reached, but not what the goal is.  It would perhaps be better to strore these setting on the Avatar.db, just to not close any options of other schemes... (same with minFee, etc)

Reply:
yes . maybe. for now there is no such a scheme req and we will use the db as a way to share data between schemes only at runtime setup ..not as a parameters setup on deployment time. though we can do that if there will be a real need.